core/src/test/scala/org/apache/spark/SparkFunSuite.scala

```diff
@@ -172,7 +173,18 @@ abstract class SparkFunSuite
     }
   }
 
-  protected def logForFailedTest(): Unit = {}
+  protected def logForFailedTest(): Unit = {
+    LocalSparkCluster.get.foreach { localCluster =>
+      val workerLogfiles = localCluster.workerLogfiles
+      if (workerLogfiles.nonEmpty) {
+        logInfo("\n\n===== EXTRA LOGS FOR THE FAILED TEST\n")
+        workerLogfiles.foreach { logFile =>
+          logInfo(s"\n----- Logfile: ${logFile.getAbsolutePath()}")
+          logInfo(FileUtils.readFileToString(logFile, "UTF-8"))
```

Comment:
This will be appended to the `unittest.log`. So in case of the example used in the PR description where one of the test from `DistributedSuite` failed (because I have  added a temporary `assert(false)`) the `core/target/unittest.log` was extended by the worker logs.